text/0009-BundleTracker-integration.md

```diff
@@ -425,33 +429,162 @@ Once a `BundleTracker` is opened, there are API methods to gain information and
 
-## How we teach this
+The `BundleTracker` uses a pointer-to-implementation pattern to hide data members and implementation details from the user. This information is contained in `BundleTrackerPrivate`, which has a one-to-one relationship with the `BundleTracker` through a `std::unique_ptr`. This construct in turn owns an instance of the `TrackedBundle`, which subclasses `BundleAbstractTracked` and handles tracking operations while the `BundleTracker` is open. Since `BundleTrackerPrivate` and `TrackedBundle` are both template classes, however, their implementations must be listed in header files which can be seen by users. A "detail" header subfolder is used as a convention to tell users that the implementation classes are subject to change and are not part of the API.
+
+The customization scheme comes from a back pointer from the `TrackedBundle` class. When a `BundleTrackerCustomizer` is provided by the user, the pointer points to it. When `BundleTrackerCustomizer` is `nullptr` when constructed, it points to the `BundleTracker` itself to use the default or overridden methods.
+
+The constructor for `BundleTracker` takes a `std::shared_ptr` to a `BundleTrackerCustomizer` instance. This guarantees that the lifetime of the `BundleTrackerCustomizer` is scoped to that of the `BundleTracker` object. This is inconsistent with the current `ServiceTracker` specification, which uses a raw pointer. In the future, the `ServiceTracker` should be amended to follow this convention.
+
+It is worth noting that the API for `BundleTracker` is different from the current implementation of the `ServiceTracker`, which still uses a raw pointer for the `ServiceTracker` constructor's `ServiceTrackerCustomizer` argument. In the future, the `ServiceTracker` API should be adapted to function like the new `BundleTracker` API to make the lifetime of the `ServiceTrackerCustomizer` more explicit.
+### Core Framework Interactions
+
+The `BundleTracker` uses the following existing constructs in its implementation:
+
+- `BundleListener`: The listener is notified about `Bundle` state change events to be processed by the `BundleTracker`.
+- `BundleContext`: The context is used for registering the listener, and listing the known bundles to add during initialization.
+- `BundleAbstractTracked`: This class is used to manage the tracking map for the `ServiceTracker`. It is reused for the `BundleTracker`.
+
+The `BundleTracker` itself is part of the core CppMicroservices framework.
+
+### Class Design
+
+The following illustrates the class diagram for the `BundleTracker`:
+
+![Class Diagram](0009-BundleTracker-integration/class_diagram.svg)
+
+#### BundleTrackerCustomizer
+
+The `BundleTrackerCustomizer` is an interface that is defined in the API section. Since all of its methods are pure virtual, it contains no implementation and exists to be subclassed by `BundleTracker` and by users.
```

Comment:
Change "and exists" to "and is intended"

Reply:
This change has been adopted.